component/common.libsonnet

```diff
@@ -66,11 +66,15 @@ local matchOrgNamespaces = matchNamespaces(
   }
 );
 
+local kyvernoPatternToRegex = function(pattern)
+  '^%s$' % std.strReplace(std.strReplace(pattern, '?', '.'), '*', '.*');
+
 {
   DefaultLabels: defaultLabels,
   FlattenSet: flattenSet,
   BypassNamespaceRestrictionsSubjects: bypassNamespaceRestrictionsSubjects,
   MatchNamespaces: matchNamespaces,
   MatchOrgNamespaces: matchOrgNamespaces,
   MatchProjectRequests: matchProjectRequests,
+  KyvernoPatternToRegex: kyvernoPatternToRegex,
```

Comment:
What is the benefit of this conversion? Wouln't it be simpler to just have the imput defined as needed? This smells like a lot of future trouble.

Reply:
It is for consistency.

https://github.com/appuio/component-appuio-cloud/blob/5fce695160f9eeb5898fb069d1e6453161bdc94a/class/defaults.yml#L20

https://github.com/appuio/component-appuio-cloud/blob/5fce695160f9eeb5898fb069d1e6453161bdc94a/class/defaults.yml#L29

We have pattern matching everywhere else. The patterns are so simple it's really easy to make regexes out of them.

If we introduce regex in one place we would have to decide if we always anchor them and somehow document all the quirks. With Kyverno patterns it's clear what is supported and what not...

In the long-term I would prefer to have a `match_pattern()` function in Kyverno.

Reply:
> In the long-term I would prefer to have a match_pattern() function in Kyverno.

Did you open a feature request at Kyvernos issue tracker?

Reply:
> > In the long-term I would prefer to have a match_pattern() function in Kyverno.
> 
> Did you open a feature request at Kyvernos issue tracker?

Not yet... I'll link it here as soon as I did.